src/molecule_plugins/podman/playbooks/create.yml

```diff
@@ -144,52 +131,56 @@
-        {{ item.extra_opts | default([]) | join(' ') }}
-        {{ item.pre_build_image | default(false) | ternary('', 'molecule_local/') }}{{ item.image }}
-        {{ (command_directives_dict | default({}))[item.name] | default('') }}
+      containers.podman.podman_container:
+        name: "{{ item.name }}"
+        cap_add: "{{ item.capabilities | default(omit) }}"
+        # it's necessary to exclude empty args as ansible modules run_command method doesn't handle this well
+        cmd_args: "{{ molecule_podman_args | reject('equalto', '') | list + item.extra_opts | default([]) }}"
+        command: "{{ (command_directives_dict | default({}))[item.name] | default('') }}"
+        detach: "{{ item.detach | default(omit) }}"
+        device: "{{ item.devices | default(omit) }}"
+        dns: "{{ item.dns_servers | default(omit) }}"
+        env: "{{ item.env | default(omit) }}"
+        etc_hosts: "{{ item.etc_hosts | default(omit) }}"
+        executable: "{{ podman_exec }}"
+        expose: "{{ item.exposed_ports | default(omit) }}"
+        hostname: "{{ item.hostname | default(omit) }}"
+        image: "{{ item.pre_build_image | default(false) | ternary('', 'molecule_local/') }}{{ item.image }}"
+        ip: "{{ item.ip | default(omit) }}"
+        network: "{{ item.network | default(omit) }}"
+        pid: "{{ item.pid_mode | default(omit) }}"
+        privileged: "{{ item.privileged | default(omit) }}"
+        publish: "{{ item.published_ports | default(omit) }}"
+        security_opt: "{{ item.security_opts | default(omit) }}"
+        systemd: "{{ item.systemd | default(omit) }}"
+        tmpfs: "{{ item.tmpfs | default(omit) }}"
+        tty: "{{ item.tty | default(omit) }}"
+        ulimits: "{{ item.ulimits | default(omit) }}"
+        volume: "{{ item.volumes | default(omit) }}"
+        state: started
       register: server
+      changed_when: false
       with_items: "{{ molecule_yml.platforms }}"
       loop_control:
         label: "{{ item.name }}"
       async: 7200
       poll: 0
-      changed_when: true
+      vars:
+        molecule_podman_args:
+          - "{% if item.cgroup_manager is defined %}--cgroup-manager={{ item.cgroup_manager }}{% endif %}"
+          - >-
+            {% if item.restart_policy is defined %}
+            --restart={{ item.restart_policy }}{% if item.restart_retries is defined %}:{{ item.restart_retries }}{% endif %}
+            {% endif %}
+          - "{% if item.storage_opt is defined %}--storage-opt={{ item.storage_opt }}{% endif %}"
+          - "{% if item.storage_driver is defined %}--storage-driver={{ item.storage_driver }}{% endif %}"
 
     - name: Wait for instance(s) creation to complete
       become: "{{ not item.item.rootless | default(omit) }}"
       ansible.builtin.async_status:
         jid: "{{ item.ansible_job_id }}"
       register: podman_jobs
+      changed_when: item is changed
```

Comment:
This line can be removed as well

Reply:
Se my previous comment because they are related.